Sources/NIOCore/Codec.swift

```diff
@@ -844,4 +851,11 @@ extension MessageToByteHandler {
             context.fireErrorCaught(error)
         }
     }
+    
+    public func flush(context: ChannelHandlerContext) {
+        context.flush()
```

Comment:
This should happen _after_ reconciling the state. `context.flush()` calls other code which may re-enter this method

Reply:
as in: just switch the lines or do `defer { context.flush() }`